rust/crd/src/security.rs

```diff
@@ -80,18 +94,11 @@ impl DruidTlsSecurity {
     /// Create a `DruidTlsSecurity` struct from the Druid custom resource and resolve
     /// all provided `AuthenticationClass` references.
     pub async fn new_from_druid_cluster(
-        client: &Client,
```

Comment:
By getting rid of the client we can write some tests :) Remember the bug in Line 107 `.and_then(|tls| tls.server_and_internal_secret_class.clone()),` ;)

Reply:
I'm looking forward to see if we can get the client out of most of the cluster_resource constructing code. That could be great for testability!